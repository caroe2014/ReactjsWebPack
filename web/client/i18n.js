// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import i18next from 'i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import Backend from 'i18next-locize-backend';

/* istanbul ignore next */
i18next
  .use(LanguageDetector)
  .use(Backend)
  .init({
    // fallback depending on user language
    fallbackLng: {
      'ja': ['ja-JP'],
      'default': ['en-US']
    },
    order: ['querystring', 'navigator'],
    lookupQuerystring: 'lng',
    allowMultiLoading: true,
    backend: {
      projectId: '838d5fce-27b5-4368-8c54-8fcb33577f9a',
      referenceLng: 'en-US',
      version: process.env.ENVIRONMENT === 'production' ? 'latest' : process.env.ENVIRONMENT || process.env.NODE_ENV
    },
    debug: false,
    detection: {
      caches: []
    },
    // have a common namespace used around the full app
    ns: ['core', `domain-${window.location.hostname}`],
    defaultNS: `domain-${window.location.hostname}`,
    fallbackNS: [`domain-${window.location.hostname}`, 'core'],

    keySeparator: false, // we use content as keys

    interpolation: {
      formatSeparator: ','
    },

    react: {
      wait: true
    }
  }, (err, t) => {
    if (err) return console.log('Something went wrong while loading.', err);
  });

export default i18next;
