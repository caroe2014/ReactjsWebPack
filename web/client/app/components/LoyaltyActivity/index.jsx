/* eslint-disable max-len */
// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.
/* eslint-disable max-len */
import React, { Component } from 'react';
import styled, { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme';
import { Flex, Text, Button } from 'rebass';
import { Trans } from 'react-i18next';
import RadioButton from 'web/client/app/components/RadioButton';
import FloatingLabelInput from 'web/client/app/components/FloatingLabelInput';
import { PHONE_NUMBER_FORMAT_REGEX,
  PHONE_NUMBER_PATTERN,
  CARD_NUMBER_FORMAT_REGEX,
  DEFAULT_LOYALTY_PIN_LENGTH
} from 'web/client/app/utils/constants';
import i18n from 'web/client/i18n';
import get from 'lodash.get';

const Container = styled(Flex)`
  align-items: center;
  max-width: 400px;
  padding: 0px 20px 0px 20px;
  flex-direction: column;
  background: #ffffff;
  width: 100%;
  ${props => props.theme.mediaBreakpoints.mobile`
    border: none;
    max-width: 100%; 
    height: auto;
    padding-left: 24px;
    padding-right: 24px;
  `};
  
`;

const LoyaltyButton = styled(Button)`
  background-color:  ${props => props.theme.colors.buttonControlColor};
  width: 100%;
  margin: 20px 0px;
  border-radius: 0;
  color: ${props => props.theme.colors.buttonTextColor};
  cursor: pointer;
  font-size: ${props => props.theme.fontSize.nm};
  disabled: ${props => props.disabled}
  font-weight: 400;
  padding: 12px;
  text-transform: uppercase;
  min-width: 240px;
  height: 40px;
  &:hover {
    box-shadow: 0px 0px 10px 2px lightgrey;
    -webkit-box-shadow: 0px 0px 10px 2px lightgrey;
    -moz-box-shadow: 0px 0px 10px 2px lightgrey;
  }
  text-align: center;
  margin: 0 auto;
  border: none;
  border-radius: 6px;
  &:focus {
    outline: none;
    box-shadow: none;
  }
  &:disabled{
    cursor: not-allowed;
  }
`;

const ButtonParent = styled(Flex)`
  justify-content: center;
  margin-top: 15px;
  width: 100%;
`;

const RadioButtonContainer = styled(Flex)`
  flex-direction: column;
  width: 100%;
  .input-field-wrapper .input-text {
    border-bottom: 1px solid #7d7d7d;
  }
  .input-field-wrapper {
    margin-bottom: 0px;
    width: 100%;
  }
`;

const HeaderText = styled(Text)`
  width: 100%;
  word-break: break-word;
  font-weight: bold;
  font-size: 24px;
  color: ${props => props.theme.colors.primaryTextColor};
  margin: 38px 0px 20px;
  text-align: center;
  padding-left: 10px;
  padding-right: 10px;
  ${props => props.theme.mediaBreakpoints.mobile`
    margin: 32px 0px 18px;
  `};
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const InstructionText = styled(Text)`
  width: 100%;
  word-break: break-word;
  font-size: 14px;
  color: ${props => props.theme.colors.secondaryTextColor};
  margin: 12px 0px 24px;
  text-align: center;
  padding-bottom: 10px;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const StyledText = styled(Text)`
  margin-left: 0px;
  color: ${props => props.theme.colors.secondaryTextColor};
  display: inline-block;
  font-size: ${props => props.theme.fontSize.nm};
  font-weight: 400;
  cursor: pointer;
  &.active {
    font-weight: 500;
  }
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const FetchAccountText = styled(StyledText)`
  margin: 0px 0px 10px 0px;
  text-align: center;
`;

class LoyaltyActivity extends Component {

  constructor (props) {
    super(props);
    const { loyaltyInfoMap, siteId, cartLoyaltyInfo } = this.props;
    const loyaltyInfoDetails = cartLoyaltyInfo || (loyaltyInfoMap && siteId && loyaltyInfoMap[siteId]);
    this.state = {
      error: '',
      pinError: '',
      showSubmitBtn: !!((loyaltyInfoDetails && loyaltyInfoDetails.loyaltyInfo)),
      selectedOption: (loyaltyInfoDetails && loyaltyInfoDetails.loyaltyInfo && loyaltyInfoDetails.selectedOption),
      formatNumber: (loyaltyInfoDetails && loyaltyInfoDetails.loyaltyInfo) || '',
      pin: ''
    };

    this.radioButtonLabel = {
      phone: i18n.t('LOYALTY_CAPTURE_PHONE'),
      card: i18n.t('LOYALTY_CAPTURE_CARD'),
      account: i18n.t('LOYALTY_CAPTURE_ACCOUNT')
    };

    this.loyaltyInputLabel = {
      phone: i18n.t('LOYALTY_CAPTURE_PHONE_NUMBER_LABEL'),
      card: i18n.t('LOYALTY_CAPTURE_CARD_NUMBER_LABEL'),
      account: i18n.t('LOYALTY_CAPTURE_ACCOUNT_NUMBER_LABEL'),
      pin: i18n.t('LOYALTY_CAPTURE_PIN_LABEL')
    };

    this.loyaltyPlaceholderLabel = {
      phone: i18n.t('LOYALTY_CAPTURE_PHONE_PLACEHOLDER'),
      card: i18n.t('LOYALTY_CAPTURE_CARD_PLACEHOLDER'),
      account: i18n.t('LOYALTY_CAPTURE_ACCOUNT_PLACEHOLDER'),
      pin: i18n.t('LOYALTY_CAPTURE_PIN_PLACEHOLDER')
    };

    this.loyaltyErrorMap = {
      phone: i18n.t('LOYALTY_CAPTURE_PHONE_ERROR'),
      card: i18n.t('LOYALTY_CAPTURE_CARD_ERROR'),
      account: i18n.t('LOYALTY_CAPTURE_ACCOUNT_ERROR'),
      pin: i18n.t('LOYALTY_CAPTURE_PIN_ERROR')
    };

    this.checkSelected = this.checkSelected.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.onCancel = this.onCancel.bind(this);
    this.handlePinChange = this.handlePinChange.bind(this);
  }

  checkSelected (type) {
    this.setState({ selectedOption: type, formatNumber: '', showSubmitBtn: false, error: '', pin: '' });
  }

  getFormatPhoneValue (inputValue) {
    const x = inputValue.replace(/\D/g, '').match(PHONE_NUMBER_FORMAT_REGEX);
    const formatNumber = !x[2]
      ? x[1] : x[1] + '-' + x[2] + (x[3] ? '-' + x[3] : '');
    return formatNumber;
  }

  getFormatNumberValue (inputValue) {
    const x = inputValue.replace(/\D/g, '').match(CARD_NUMBER_FORMAT_REGEX);
    const formatNumber = !x[2] ? x[1] : '';
    return formatNumber;
  }

  handleChange (propertyName, inputValue) {
    const formatNumber = propertyName === 'phone'
      ? this.getFormatPhoneValue(inputValue)
      : this.getFormatNumberValue(inputValue);
    let showSubmitBtn = !(!formatNumber ||
      (propertyName === 'phone' && formatNumber.replace(/[-\s]+/g, '').length < 10));
    this.setState({
      formatNumber,
      showSubmitBtn,
      error: !inputValue ? '' : this.state.error
    });
  }

  handlePinChange (propertyname, inputValue) {
    const { loyaltyPaymentConfiguration } = this.props;
    const configuredPinLength = parseInt(get(loyaltyPaymentConfiguration, 'accountEntry.pinLength') || DEFAULT_LOYALTY_PIN_LENGTH);

    if (inputValue.length <= configuredPinLength) {
      this.setState({
        pin: inputValue
      });
    }
  }

  onSubmit () {
    const { formatNumber, selectedOption, pin } = this.state;
    const { handleLoyaltyInfo, displayProfileId, siteId, loyaltyPaymentConfiguration } = this.props;
    const configuredPinLength = parseInt(get(loyaltyPaymentConfiguration, 'accountEntry.pinLength') || DEFAULT_LOYALTY_PIN_LENGTH);

    if (!formatNumber || (loyaltyPaymentConfiguration && !pin) || (selectedOption === 'phone' && formatNumber.replace(/[-\s]+/g, '').length < 10)) {
      this.setState({error: this.loyaltyErrorMap[selectedOption]});
      return;
    } else if (loyaltyPaymentConfiguration && pin.length !== configuredPinLength) {
      this.setState({pinError: `Pin should be ${configuredPinLength} digits`});
      return;
    }
    this.setState({
      error: '',
      pinError: ''
    });
    const loyaltyInquiryInfo = {
      formatNumber: formatNumber,
      selectedOption,
      pin: window.btoa(pin),
      siteId,
      displayProfileId
    };
    handleLoyaltyInfo(loyaltyInquiryInfo);
  }

  onCancel () {
    const { cancelLoyaltyInfo } = this.props;
    cancelLoyaltyInfo();
  }

  componentDidMount () {
    document.getElementById('loyalty-activity-header') &&
      document.getElementById('loyalty-activity-header').focus();
  }

  render () {
    const { loyaltyDetailsAccounts, fetching, fetchAccount, loyaltyDetails, loyaltyPaymentConfiguration } = this.props;
    const { error, formatNumber, selectedOption, showSubmitBtn, pin, pinError } = this.state;

    const filteredAccountDetails = loyaltyDetailsAccounts
      ? loyaltyDetailsAccounts.filter(accountItems => accountItems.enabled) : [];

    return (
      <ThemeProvider theme={theme}>
        <Container className='loyalty-activity-container'>
          <HeaderText className='loyalty-activity-header' id='loyalty-activity-header'
            tabIndex={0} role='header' aria-label={(loyaltyDetails && loyaltyDetails.header) || i18n.t('LOYALTY_CAPTURE_POINTS')}>
            {(loyaltyDetails && loyaltyDetails.header) || <Trans i18nKey='LOYALTY_CAPTURE_POINTS'/>}
          </HeaderText>
          {loyaltyDetails && loyaltyDetails.instructionText &&
            <InstructionText aria-label={loyaltyDetails.instructionText}
              tabIndex={0}
              className='instruction-text'>
              {loyaltyDetails.instructionText}
            </InstructionText>}
          <RadioButtonContainer role='radiogroup' className='loyalty-activity-radio-div'>
            {
              filteredAccountDetails.length > 0 &&
               filteredAccountDetails.map((accountItems, index) => (
                 <React.Fragment key={`account-${index}`}>
                   <RadioButton
                     className={`${accountItems.id}-radio`}
                     classDesc={`${accountItems.id}`}
                     label={this.radioButtonLabel[accountItems.id]}
                     type={accountItems.id}
                     capacityText
                     selectedOption={this.checkSelected}
                     selected={selectedOption === accountItems.id}
                     ariaChecked={selectedOption === accountItems.id}
                     tabIndex={0}
                     ariaLabel={this.radioButtonLabel[accountItems.id]}
                   />
                 </React.Fragment>
               ))}
            {
              selectedOption &&
              <React.Fragment>
                <FloatingLabelInput
                  ariaLabel={this.loyaltyInputLabel[selectedOption]}
                  propertyName={selectedOption}
                  label={this.loyaltyInputLabel[selectedOption]}
                  value={formatNumber}
                  error={error}
                  callBack={this.handleChange}
                  clearIcon
                  inputType={'tel'}
                  pattern={PHONE_NUMBER_PATTERN}
                  loyalty
                  disabled={fetching}
                  placeHolder={this.loyaltyPlaceholderLabel[selectedOption]}
                  tabIndex={0}
                />
              </React.Fragment>
            }
            {
              selectedOption && loyaltyPaymentConfiguration &&
              <FloatingLabelInput
                ariaLabel={this.loyaltyInputLabel['pin']}
                propertyName='pin'
                label={this.loyaltyInputLabel['pin']}
                value={this.state.pin}
                error={pinError}
                inputType='password'
                callBack={this.handlePinChange}
                clearIcon
                loyalty
                tabIndex={0}
              />
            }

          </RadioButtonContainer>
          {fetchAccount && <FetchAccountText tabIndex={0}
            aria-label={i18n.t('LOYALTY_FETCHING_ACCOUNT_LABEL')}>
            <Trans i18nKey='LOYALTY_FETCHING_ACCOUNT_LABEL'/>
          </FetchAccountText>}
          {selectedOption && showSubmitBtn && (!loyaltyPaymentConfiguration || pin.length > 0) &&
            <ButtonParent className='button-parent'>
              <LoyaltyButton
                className='submit-button'
                onClick={fetching ? this.onCancel : this.onSubmit}
                role='button'
                tabIndex={0}
                aria-label={fetching ? i18n.t('MODAL_CANCEL') : i18n.t('LOYALTY_CAPTURE_SUBMIT')}>
                {<Trans i18nKey={fetching ? 'MODAL_CANCEL' : 'LOYALTY_CAPTURE_SUBMIT'}/>}
              </LoyaltyButton>
            </ButtonParent>}
        </Container>

      </ThemeProvider>
    );
  }
}

export default LoyaltyActivity;
