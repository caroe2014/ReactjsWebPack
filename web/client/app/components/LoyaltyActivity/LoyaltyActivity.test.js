import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme.js';
import i18n from 'i18next';
import LoyaltyActivity from '.';
import jsdom from 'jsdom';
import { createMemoryHistory } from 'history';

configure({ adapter: new Adapter() });

/* global describe, it, expect, beforeEach, jest */
/* eslint-disable max-len */
global.window = new jsdom.JSDOM().window;
global.document = window.document;
global.scrollTo = jest.fn();

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));
jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  withTranslation: () => Component => {
    Component.defaultProps = { ...Component.defaultProps, t: (key) => key };
    return Component;
  }
}));

jest.mock('web/client/app/reduxpages/ConnectedComponents', () => {
  return {
    ConnectedLoyaltyProcess: <div/>
  };
});

jest.mock('react-i18next', () => ({
  Trans: 'div'
}));

describe('LoyaltyActivity', () => {

  let wrapper;
  let loyaltyDetailsAccounts = [
    {
      'id': 'phone',
      'enabled': true
    },
    {
      'id': 'card',
      'enabled': true
    },
    {
      'id': 'account',
      'enabled': false
    }
  ];
  let fetching = false;
  const history = createMemoryHistory('/payments');
  let list = [{id: 1001, isLoyaltyEnabled: true}];

  beforeEach(() => {
    wrapper = shallow(<LoyaltyActivity loyaltyDetailsAccounts={loyaltyDetailsAccounts} fetching={false} history={history}/>);
  });

  it('does a snapshot test', () => {
    const snapwrapper = mount(<ThemeProvider theme={theme}><LoyaltyActivity loyaltyDetailsAccounts={loyaltyDetailsAccounts} fetching={false}/></ThemeProvider>);
    expect(toJson(snapwrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(wrapper.exists()).toBe(true);
  });

  // describe('when fetching is true', () => {
  //   it('should render without throwing an error', () => {
  //     const fetchWrapper = shallow(<LoyaltyActivity loyaltyDetailsAccounts={loyaltyDetailsAccounts} fetching history={history}/>);
  //     expect(fetchWrapper.exists()).toBe(true);
  //   });
  //   it('should trigger onCancel when cancel button is clicked', () => {
  //     let mockFn = jest.fn();
  //     const fetchWrapper = shallow(<LoyaltyActivity loyaltyDetailsAccounts={loyaltyDetailsAccounts} cancelLoyaltyInfo={mockFn} fetching history={history}/>);
  //     const spy = jest.spyOn(fetchWrapper.instance(), 'onCancel');
  //     fetchWrapper.instance().forceUpdate();
  //     fetchWrapper.find('.submit-button').simulate('click');
  //     expect(spy).toHaveBeenCalled();
  //     spy.mockClear();
  //   });
  // });

  // describe('when fetching is false', () => {
  //   let mockFn = jest.fn();
  //   it('should trigger onSubmit when Submit button is clicked', () => {
  //     const fetchWrapper = shallow(<LoyaltyActivity loyaltyDetailsAccounts={loyaltyDetailsAccounts} handleLoyaltyInfo={mockFn} fetching={false} history={history}/>);
  //     const spy = jest.spyOn(fetchWrapper.instance(), 'onSubmit');
  //     fetchWrapper.instance().forceUpdate();
  //     fetchWrapper.find('.submit-button').simulate('click');
  //     expect(spy).toHaveBeenCalled();
  //     spy.mockClear();
  //   });
  //   it('should trigger onSubmit when Submit button is clicked ', () => {
  //     const fetchWrapper = shallow(<LoyaltyActivity loyaltyDetailsAccounts={loyaltyDetailsAccounts} handleLoyaltyInfo={mockFn} fetching={false} history={history}/>);
  //     const spy = jest.spyOn(fetchWrapper.instance(), 'onSubmit');
  //     fetchWrapper.setState({ formatNumber: '9876527282', selectedOption: 'phone' });
  //     fetchWrapper.instance().forceUpdate();
  //     fetchWrapper.update();
  //     fetchWrapper.find('.submit-button').simulate('click');
  //     expect(spy).toHaveBeenCalled();
  //     spy.mockClear();
  //   });
  // });

  it('should render without throwing an error when loyaltyDetailsAccounts is not available', () => {
    const fetchWrapper = shallow(<LoyaltyActivity fetching loyaltyDetailsAccounts={loyaltyDetailsAccounts} history={history}/>);
    expect(fetchWrapper.exists()).toBe(true);
  });

  it('should render without throwing an error when loyaltyInfoMap  is  available', () => {
    const loyaltyInfoMap = {
      1001: {
        test: 'test',
        loyaltyInfo: {
          test: 'test',
          selectedOption: 'phone'
        }
      }
    };
    const fetchWrapper = shallow(
      <LoyaltyActivity
        fetching
        loyaltyDetailsAccounts={loyaltyDetailsAccounts}
        loyaltyInfoMap={loyaltyInfoMap}
        siteId={1001}
        history={history}/>
    );
    expect(fetchWrapper.exists()).toBe(true);
  });

  // it('should render only one option with default enabled', () => {
  //   let loyaltyDetailsAccounts = [
  //     {
  //       'id': 'phone',
  //       'enabled': true
  //     },
  //     {
  //       'id': 'card',
  //       'enabled': false
  //     },
  //     {
  //       'id': 'account',
  //       'enabled': false
  //     }
  //   ];
  //   const newWrapper = shallow(<LoyaltyActivity loyaltyDetailsAccounts={loyaltyDetailsAccounts} fetching={false} history={history}/>);
  //   // const newWrapper = mount(<ThemeProvider theme={theme}><LoyaltyActivity loyaltyDetailsAccounts={loyaltyDetailsAccounts} fetching={false}/></ThemeProvider>);
  //   expect(newWrapper.find('.option_desc').length).toBe(1);
  // });

  it('should render only one option with default enabled', () => {
    let loyaltyDetailsAccounts = [
      {
        'id': 'phone',
        'enabled': true
      },
      {
        'id': 'card',
        'enabled': false
      },
      {
        'id': 'account',
        'enabled': false
      }
    ];
    const newWrapper = mount(<ThemeProvider theme={theme}><LoyaltyActivity loyaltyDetailsAccounts={loyaltyDetailsAccounts} fetching={false}/></ThemeProvider>);
    expect(newWrapper.exists()).toBe(true);
  });

});
