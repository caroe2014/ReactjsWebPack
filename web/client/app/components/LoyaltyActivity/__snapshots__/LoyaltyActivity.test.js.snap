// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoyaltyActivity does a snapshot test 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "addOnTab": "#347CFF",
        "addToCart": "#347CFF",
        "addonBg": "#EEEEEE",
        "backgroundGrey": "#E6E6E6",
        "black": "#000000",
        "close": "red",
        "contextBackground": "#f0f0f3",
        "darkGrey": "#464646",
        "disableBackground": "#E4E4E4",
        "disableButton": "#D1D1D1",
        "error": "#cc0000",
        "footer": "#48a942",
        "glutenFree": "#8d7356",
        "healthy": "#359919",
        "hint": "#a6a6a6",
        "inputBottomBorder": "lightgray",
        "itemBorder": "lightgrey",
        "itemTile": "#3C474C0",
        "light": "white",
        "loader": "#ababab",
        "loginBackground": "#303030",
        "loginBoxColor": "#434149",
        "loginField": "#AAA7AC",
        "loginFormBackground": "#484848",
        "navigationMenu": "#48a942",
        "navigationMenuHover": "cornflowerblue",
        "new": "#ff6000",
        "open": "#48a942",
        "paymentSuccessTitle": "#505050",
        "placeHolder": "#909090",
        "primary": "#347CFF",
        "primaryButton": "#347CFF",
        "primaryText": "#383838",
        "secondaryText": "#a6a6a6",
        "spicy": "#ed1c24",
        "text": "black",
        "textGrey": "#505050",
        "tile": "#505050",
        "validationError": "#cc0000",
        "vegan": "#0d4a22",
        "vegetarian": "#0d4a22",
      },
      "fontFamily": "Roboto",
      "fontSize": Object {
        "lg": "1.8rem",
        "md": "1.3rem",
        "nm": "1rem",
        "sm": "0.75rem",
        "xl": "2.4rem",
      },
      "fontSizeFromProps": [Function],
      "mediaBreakpoints": Object {
        "desktop": [Function],
        "mobile": [Function],
        "tablet": [Function],
      },
      "nav": Object {
        "height": "70px",
      },
      "receiptFontFamily": "Roboto",
      "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
      "spacing": Object {
        "lg": "4em",
        "md": "2em",
        "nm": "1em",
        "none": "0em",
        "sm": "0.5em",
        "xl": "8em",
      },
    }
  }
>
  <LoyaltyActivity
    fetching={false}
    loyaltyDetailsAccounts={
      Array [
        Object {
          "enabled": true,
          "id": "phone",
        },
        Object {
          "enabled": true,
          "id": "card",
        },
        Object {
          "enabled": false,
          "id": "account",
        },
      ]
    }
  >
    <ThemeProvider
      theme={
        Object {
          "colors": Object {
            "addOnTab": "#347CFF",
            "addToCart": "#347CFF",
            "addonBg": "#EEEEEE",
            "backgroundGrey": "#E6E6E6",
            "black": "#000000",
            "close": "red",
            "contextBackground": "#f0f0f3",
            "darkGrey": "#464646",
            "disableBackground": "#E4E4E4",
            "disableButton": "#D1D1D1",
            "error": "#cc0000",
            "footer": "#48a942",
            "glutenFree": "#8d7356",
            "healthy": "#359919",
            "hint": "#a6a6a6",
            "inputBottomBorder": "lightgray",
            "itemBorder": "lightgrey",
            "itemTile": "#3C474C0",
            "light": "white",
            "loader": "#ababab",
            "loginBackground": "#303030",
            "loginBoxColor": "#434149",
            "loginField": "#AAA7AC",
            "loginFormBackground": "#484848",
            "navigationMenu": "#48a942",
            "navigationMenuHover": "cornflowerblue",
            "new": "#ff6000",
            "open": "#48a942",
            "paymentSuccessTitle": "#505050",
            "placeHolder": "#909090",
            "primary": "#347CFF",
            "primaryButton": "#347CFF",
            "primaryText": "#383838",
            "secondaryText": "#a6a6a6",
            "spicy": "#ed1c24",
            "text": "black",
            "textGrey": "#505050",
            "tile": "#505050",
            "validationError": "#cc0000",
            "vegan": "#0d4a22",
            "vegetarian": "#0d4a22",
          },
          "fontFamily": "Roboto",
          "fontSize": Object {
            "lg": "1.8rem",
            "md": "1.3rem",
            "nm": "1rem",
            "sm": "0.75rem",
            "xl": "2.4rem",
          },
          "fontSizeFromProps": [Function],
          "mediaBreakpoints": Object {
            "desktop": [Function],
            "mobile": [Function],
            "tablet": [Function],
          },
          "nav": Object {
            "height": "70px",
          },
          "receiptFontFamily": "Roboto",
          "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
          "spacing": Object {
            "lg": "4em",
            "md": "2em",
            "nm": "1em",
            "none": "0em",
            "sm": "0.5em",
            "xl": "8em",
          },
        }
      }
    >
      <Styled(Flex)
        className="loyalty-activity-container"
      >
        <Flex
          className="loyalty-activity-container sc-lhVmIH eRybNx"
        >
          <Box
            className="loyalty-activity-container sc-lhVmIH eRybNx sc-bwzfXH hKiLMS"
          >
            <div
              className="loyalty-activity-container sc-lhVmIH eRybNx sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
            >
              <Styled(Text)
                aria-label="LOYALTY_CAPTURE_POINTS"
                className="loyalty-activity-header"
                id="loyalty-activity-header"
                role="header"
                tabIndex={0}
              >
                <Text
                  aria-label="LOYALTY_CAPTURE_POINTS"
                  className="loyalty-activity-header sc-kTUwUJ bPPlLA"
                  id="loyalty-activity-header"
                  m={0}
                  role="header"
                  tabIndex={0}
                >
                  <ForwardRef
                    aria-label="LOYALTY_CAPTURE_POINTS"
                    className="loyalty-activity-header sc-kTUwUJ bPPlLA sc-gqjmRU kKsieB"
                    id="loyalty-activity-header"
                    m={0}
                    role="header"
                    tabIndex={0}
                  >
                    <div
                      aria-label="LOYALTY_CAPTURE_POINTS"
                      className="loyalty-activity-header sc-kTUwUJ bPPlLA sc-gqjmRU kKsieB"
                      id="loyalty-activity-header"
                      role="header"
                      tabIndex={0}
                    >
                      <div
                        i18nKey="LOYALTY_CAPTURE_POINTS"
                      />
                    </div>
                  </ForwardRef>
                </Text>
              </Styled(Text)>
              <Styled(Flex)
                className="loyalty-activity-radio-div"
                role="radiogroup"
              >
                <Flex
                  className="loyalty-activity-radio-div sc-jtRfpW bpOQwk"
                  role="radiogroup"
                >
                  <Box
                    className="loyalty-activity-radio-div sc-jtRfpW bpOQwk sc-bwzfXH hKiLMS"
                    role="radiogroup"
                  >
                    <div
                      className="loyalty-activity-radio-div sc-jtRfpW bpOQwk sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
                      role="radiogroup"
                    >
                      <RadioButton
                        ariaChecked={false}
                        ariaLabel="LOYALTY_CAPTURE_PHONE"
                        capacityText={true}
                        classDesc="phone"
                        className="phone-radio"
                        label="LOYALTY_CAPTURE_PHONE"
                        selected={false}
                        selectedOption={[Function]}
                        tabIndex={0}
                        type="phone"
                      >
                        <Styled(Flex)
                          aria-checked={false}
                          aria-label="LOYALTY_CAPTURE_PHONE"
                          capacitytext={1}
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          role="radio"
                          tabIndex={0}
                        >
                          <Flex
                            aria-checked={false}
                            aria-label="LOYALTY_CAPTURE_PHONE"
                            capacitytext={1}
                            className="sc-uJMKN isoDuX"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="radio"
                            tabIndex={0}
                          >
                            <Box
                              aria-checked={false}
                              aria-label="LOYALTY_CAPTURE_PHONE"
                              capacitytext={1}
                              className="sc-uJMKN isoDuX sc-bwzfXH hKiLMS"
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              role="radio"
                              tabIndex={0}
                            >
                              <div
                                aria-checked={false}
                                aria-label="LOYALTY_CAPTURE_PHONE"
                                className="sc-uJMKN isoDuX sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
                                onClick={[Function]}
                                onKeyDown={[Function]}
                                role="radio"
                                tabIndex={0}
                              >
                                <i
                                  className="fa fa-circle-thin phone radio-o"
                                />
                                <Styled(Text)
                                  capacitytext={1}
                                  className="option_desc"
                                  id="phone"
                                >
                                  <Text
                                    capacitytext={1}
                                    className="option_desc sc-bbmXgH bzKUMN"
                                    id="phone"
                                    m={0}
                                  >
                                    <ForwardRef
                                      capacitytext={1}
                                      className="option_desc sc-bbmXgH bzKUMN sc-gqjmRU kKsieB"
                                      id="phone"
                                      m={0}
                                    >
                                      <div
                                        capacitytext={1}
                                        className="option_desc sc-bbmXgH bzKUMN sc-gqjmRU kKsieB"
                                        id="phone"
                                      >
                                        LOYALTY_CAPTURE_PHONE
                                      </div>
                                    </ForwardRef>
                                  </Text>
                                </Styled(Text)>
                              </div>
                            </Box>
                          </Flex>
                        </Styled(Flex)>
                      </RadioButton>
                      <RadioButton
                        ariaChecked={false}
                        ariaLabel="LOYALTY_CAPTURE_CARD"
                        capacityText={true}
                        classDesc="card"
                        className="card-radio"
                        label="LOYALTY_CAPTURE_CARD"
                        selected={false}
                        selectedOption={[Function]}
                        tabIndex={0}
                        type="card"
                      >
                        <Styled(Flex)
                          aria-checked={false}
                          aria-label="LOYALTY_CAPTURE_CARD"
                          capacitytext={1}
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          role="radio"
                          tabIndex={0}
                        >
                          <Flex
                            aria-checked={false}
                            aria-label="LOYALTY_CAPTURE_CARD"
                            capacitytext={1}
                            className="sc-uJMKN isoDuX"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="radio"
                            tabIndex={0}
                          >
                            <Box
                              aria-checked={false}
                              aria-label="LOYALTY_CAPTURE_CARD"
                              capacitytext={1}
                              className="sc-uJMKN isoDuX sc-bwzfXH hKiLMS"
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              role="radio"
                              tabIndex={0}
                            >
                              <div
                                aria-checked={false}
                                aria-label="LOYALTY_CAPTURE_CARD"
                                className="sc-uJMKN isoDuX sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
                                onClick={[Function]}
                                onKeyDown={[Function]}
                                role="radio"
                                tabIndex={0}
                              >
                                <i
                                  className="fa fa-circle-thin card radio-o"
                                />
                                <Styled(Text)
                                  capacitytext={1}
                                  className="option_desc"
                                  id="card"
                                >
                                  <Text
                                    capacitytext={1}
                                    className="option_desc sc-bbmXgH bzKUMN"
                                    id="card"
                                    m={0}
                                  >
                                    <ForwardRef
                                      capacitytext={1}
                                      className="option_desc sc-bbmXgH bzKUMN sc-gqjmRU kKsieB"
                                      id="card"
                                      m={0}
                                    >
                                      <div
                                        capacitytext={1}
                                        className="option_desc sc-bbmXgH bzKUMN sc-gqjmRU kKsieB"
                                        id="card"
                                      >
                                        LOYALTY_CAPTURE_CARD
                                      </div>
                                    </ForwardRef>
                                  </Text>
                                </Styled(Text)>
                              </div>
                            </Box>
                          </Flex>
                        </Styled(Flex)>
                      </RadioButton>
                    </div>
                  </Box>
                </Flex>
              </Styled(Flex)>
            </div>
          </Box>
        </Flex>
      </Styled(Flex)>
    </ThemeProvider>
  </LoyaltyActivity>
</ThemeProvider>
`;
